src/Shared/Video.h

```diff
@@ -26,6 +26,10 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 extern GXRModeObj *rmode;
 extern void *xfb;
 
+// upstream these macros into libogc imo to complement VI_TVMODE (https://libogc.devkitpro.org/video__types_8h.html)
```

Comment:
* put #ifdef's around these so that when they are upstreamed they don't cause a conflict.
* add an include to `video__types.h` as i might be adding them there if this works.
* add a Vi_TVMODE_HASFMT which does the == , which i will add too if this works, and use it for comparisons. same with the mode define

Reply:
will do; on the third bullet, could you please explain which syntax you want exactly and why you prefer it to just using ==?

Reply:
just easier to read imo, `if(HasFormat(videoMode, mode))` is just easier to read to me :D

Reply:
that's fair yeah. also, `video_types.h` isn't included anywhere in the codebase so may not be necessary; wdyt?

Reply:
not right now, no. but i will be putting the defines there :)

Reply:
ye, maybe do `IS`. HAS is more if multiple are possible like in a list or something